### ISSUE TITLE
[Bug]: Group items positioning behaving weirdly with FixedLayout

### ISSUE THREAD

**Issue by mvuckovic** [fabricjs/fabric.js#10035]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.2

### In What environments are you experiencing the problem?

Firefox

### Node Version (if applicable)

None

### Link To Reproduction

https://codesandbox.io/p/sandbox/epic-bouman-8myz6h?file=%2Fsrc%2Findex.ts

### Steps To Reproduce

1. Open reproduction link
2. 
3.

### Expected Behavior

I expected that 
- Group with FixedLayout behaves the same regardless if items were added initially or later 
- Items to be aligned top/left in regards to group position

Basically I'm trying to create a *fixed* container where you can add items inside and they'll be positioned relative to that container.

### Actual Behavior

1. Group with FixedLayout and items added initially has some kind of additional top/left offset 
2. Group without FixedLayout and items added initially behaves as expected
3. Group with FixedLayout and items added later seems to position items relative to canvas itself

**Comment by mvuckovic** [comment]
Hi and thanks for responding :)

I'm trying to implement some kind of "grid manager" show in the image below
![image](https://github.com/user-attachments/assets/c9fd1667-aa06-4f7b-80ca-653d225c5535)

Where you can add Cells which live inside area/row/column matrix. 
Cells can then have text/images/icons which are positioned relative to cell itself.

It made sense to me to define each cell as a group that contains a cliparea and can be positioned/resized by the grid-manager.

Does that mean I should avoid using groups as a container for items? 

Kind regards.

**Comment by asturur** [comment]
i have no idea of the performances but with that graphical example i would look into writing a custom layout manager for group that can handle colored rect and behave as you want.

Once you have that done i would swap the rects with groups with the clippath layout ( fixed layout will have some graphical glitch related to caching ) that can have the same shape of rects but can contain the items you want.

I never built this, so as first step i would divide the work in 2 parts

You can add the index of your grid manager to object and have the layout manager inspect those and take decision on position and size.

The layout manager was built to customize this kind of behaviour, then if is ready or not for that i can't tell

**Comment by mvuckovic** [comment]
Sorry I wasn't clear initially; I've already build Grid/Layout Manager and it's working fine and performance seems very ok (I haven't tried yet adding a bunch of images into cells)

My confusion was about positioning of group items. I've read _somewhere_ in the documentation that group items are positioned relative to the group but that's not what I'm seeing in practice when group has a fixed top/left/width/height + FixedLayout set.

It seems that the solution for now is to always add group top/left when calculation position of items inside as everything is relative to canvas.

**Comment by asturur** [comment]
The generic behaviour of the group is to keep position of the object unchanged when added from the canvas.
When the group has a fixed size the top / left corner of the group is the negative half of the size.
Is there some kind of issue that makes positioning not behaving as expected?

the 0,0 in the group is always the center
